openmc/model/funcs.py

```diff
@@ -110,6 +111,140 @@ def borated_water(boron_ppm, temperature=293., pressure=0.1013, temp_unit='K',
+                   **{axis + '0': value})
+    def cylinder(axis, name, ax1, val1, ax2, val2, r):
+        cls = globals()['{}Cylinder'.format(axis.upper())]
+        return cls(name='{} {}'.format(name, axis),
+                   boundary_type=boundary_type,
+                   **{ax1 + '0': val1,
+                      ax2 + '0': val2,
+                      'r': r})
+
+    if axis == 'x':
+        x1, x2 = 'y', 'z'
+        axcoord = 0
+        axcoord1 = 1
+        axcoord2 = 2
+    elif axis == 'y':
+        x1, x2 = 'x', 'z'
+        axcoord = 1
+        axcoord1 = 0
+        axcoord2 = 2
+    else:
+        x1, x2 = 'x', 'y'
+        axcoord = 2
+        axcoord1 = 0
+        axcoord2 = 1
+
+    # Get torus class corresponding to given axis
+    tor = globals()['{}Torus'.format(axis.upper())]
+
+    # Create cylindricalr region
+    min_h = plane(axis, 'minimum', -height/2 + origin[axcoord])
+    max_h = plane(axis, 'maximum', height/2 + origin[axcoord])
+    radial = cylinder(axis, 'outer', x1, origin[axcoord1], x2, origin[axcoord2], r)
+
+    if boundary_type == 'periodic':
+        min_h.periodic_surface = max_h
+        max_h.periodic_surface = min_h
+    prism = +min_h & -max_h & -radial
+
+    # Handle rounded corners if given
+    if upper_edge_radius > 0. or lower_edge_radius > 0.:
+        if boundary_type == 'periodic':
+            raise ValueError('Periodic boundary conditions not permitted when '
+                             'rounded corners are used.')
+
+        corners_upper = None
+        corners_lower = None
+        args = {'boundary_type': boundary_type}
+        args[x1 + '0'] = origin[axcoord1]
+        args[x2 + '0'] = origin[axcoord2]
+        if upper_edge_radius > 0.:
+            args['a'] = r - upper_edge_radius
+            args['b'] = upper_edge_radius
+            args['c'] = upper_edge_radius
+            args[axis + '0'] = origin[axcoord] + height/2 - upper_edge_radius
+            tor_upper_max = tor(name='{} max'.format(axis), **args)
+
+            axis_upper_min = plane(axis, 'upper min', height/2 + origin[axcoord] - upper_edge_radius)
+            cyl_upper_min = cylinder(axis, 'upper min', x1, origin[axcoord1], x2, origin[axcoord2], r - upper_edge_radius)
+
+            corners_upper = (+cyl_upper_min & +tor_upper_max & + axis_upper_min)
+
+        if lower_edge_radius > 0.:
+            args['a'] = r - lower_edge_radius
+            args['b'] = lower_edge_radius
+            args['c'] = lower_edge_radius
+            args[axis + '0'] = origin[axcoord] - height/2 + lower_edge_radius
+            tor_lower_min = tor(name='{} min'.format(axis), **args)
+
+            axis_lower_max = plane(axis, 'lower max', origin[axcoord] - height/2 + lower_edge_radius)
+            cyl_lower_min = cylinder(axis, 'lower min', x1, origin[axcoord1], x2, origin[axcoord2], r - lower_edge_radius)
+
+            corners_lower = (+cyl_lower_min & +tor_lower_min & - axis_lower_max)
+
```

Comment:
It seems as though there is an opportunity to reuse at least some of this code in a modular way (e.g. a function or loop) rather than cut/paste for the upper/lower.  The benefit of this kind of reuse is for consistency/maintainability and not for performance.